File: 백준/Gold/2467. 용액/용액.py
import sys
input = sys.stdin.readline

n = int(input())
arr = list(map(int,input().split()))
result = [arr[0],arr[n-1]]

minSum = abs(sum(result))

s= 0
e = n-1

while(s<e):
  diff = arr[s] + arr[e]
  if(abs(diff) < minSum):
    minSum = abs(diff)
    result = [arr[s],arr[e]]
  if(diff > 0):
    e -=1
  elif(diff <0):
    s+=1
  else:
    break

print( ' '.join(list(map(str,result))))

File: 백준/Gold/2467. 용액/용액.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const N = Number(input[0])
const numbers = input[1].split(" ").map(Number)
let start = 0
let end = N - 1
const answer = [numbers[start], numbers[end]]

let diff = Math.abs(numbers[start] + numbers[end])

while (start < end-1) {

  if (diff == 0) break
  const nextStart = Math.abs(numbers[start + 1] + numbers[end])
  const nextEnd = Math.abs(numbers[start] + numbers[end - 1])

  if (nextStart < nextEnd) {
    if (diff > nextStart) {
      answer[0] = numbers[start + 1]
      answer[1] = numbers[end]
      diff = nextStart
    }
    start += 1
  } else {
    if (diff > nextEnd) {
      diff = nextEnd
      answer[0] = numbers[start]
      answer[1] = numbers[end - 1]
    }
    end -= 1
  }

}
console.log(answer.join(" "))
